revocation/revocation.go

```diff
@@ -0,0 +1,249 @@
+package revocation
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"golang.org/x/crypto/ocsp"
+)
+
+// Revocation is an internal struct used for revocation checking
+type revocation struct {
+	httpClient      *http.Client
+	wg              sync.WaitGroup
+	mergeErrorsFunc func(errors []error) error
+}
+
+// NewRevocation constructs a revocation object and substitutes default values for any that are passed as nil
+func NewRevocation(httpClient *http.Client) *revocation {
+	if httpClient != nil {
+		return &revocation{httpClient: httpClient, wg: sync.WaitGroup{}, mergeErrorsFunc: defaultMergeErrors}
+	}
+	return &revocation{httpClient: http.DefaultClient, wg: sync.WaitGroup{}, mergeErrorsFunc: defaultMergeErrors}
```

Comment:
I don't see the harm here in defaulting. IMO it provides a better user experience by not requiring users to define a specific http client if they are going to use the default client anyways

Reply:
We are not doing anything specific so even caller can pass `http.DefaultClient` as a prameter. Also forcing caller to pass http client means they need to make conscious call about timeouts.

Reply:
I think this boils down to what we would prefer for the user experience. Is it better to allow users to defer their choice to our default, or is it better to force a user to choose a timeout. My opinion is that a user who knows what timeout they want will still be able to pass it in either situation. However, a user who is not sure what would be a good timeout to use can save the time of having to look into it themselves by simply using our default timeout. I believe this is beneficial to the customer experience for less experienced users.

Instead of a default client, would it be better to specify a client with a timeout of 5 seconds? Or do you still believe that forcing the user to pass a client is the correct option.

Reply:
My opinion is that a user who knows what timeout they want will still be able to pass it in either situation. However, a user who is not sure what would be a good timeout to use can save the time of having to look into it themselves by simply using our default timeout

> What is default timeout? core-go should rarely have defaults. If someone is directly using core-go means they shoudl understand the intricacies of golang, ocsp, etc.

Reply:
Okay, that makes sense. I wasn't sure who the users are of core-go